addons/project/tests/test_project_flow.py

```diff
@@ -117,3 +120,74 @@ def test_subtask_process(self):
         })
 
         self.assertEqual(parent_task.partner_id, child_task.partner_id, "Subtask partner should not change when changing project")
+
+    def test_rating(self):
+        """Check if rating works correctly even when task is changed from project A to project B"""
+        Task = self.env['project.task'].with_context({'tracking_disable': True})
+        first_task = Task.create({
+            'name': 'first task',
+            'user_id': self.user_projectuser.id,
+            'project_id': self.project_pigs.id,
+            'partner_id': self.partner_2.id,
+            'planned_hours': 0,
+        })
+
+        self.assertEqual(first_task.rating_count, 0, "Task should have no rating associated with it")
+
+        Rating = self.env['rating.rating']
+        rating_good = Rating.create({
+            'res_model_id': self.env['ir.model']._get('project.task').id,
+            'res_id': first_task.id,
+            'parent_res_model_id': self.env['ir.model']._get('project.project').id,
+            'parent_res_id': self.project_pigs.id,
+            'rated_partner_id': self.partner_2.id,
+            'partner_id': self.partner_2.id,
+            'rating': 10,
+            'consumed': True,
```

Comment:
Maybe we should test sending a rating invitation (`consumed=False` implies that `rating_count` is still zero),  then consume it by setting a real value.